url27a7ae3508_serkan_ozal_ocean_of_memories_tgz-pJ8-com_zeroturnaround_rebellabs_oceanofmemories_article1_objectlayout_ObjectMemoryLayoutDemoJ8/src/com/zeroturnaround/rebellabs/oceanofmemories/article2/benchmark/memorymappedfile/MemoryMappedFileBasedOffHeapBenchmarkWorker.java

```diff
@@ -0,0 +1,61 @@
+package com.zeroturnaround.rebellabs.oceanofmemories.article2.benchmark.memorymappedfile;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.RandomAccessFile;
+import java.nio.ByteBuffer;
+import java.nio.MappedByteBuffer;
+import java.nio.channels.FileChannel;
+
+import com.zeroturnaround.rebellabs.oceanofmemories.article2.benchmark.ByteBufferBasedOffHeapBenchmarkWorker;
+
+public class MemoryMappedFileBasedOffHeapBenchmarkWorker extends ByteBufferBasedOffHeapBenchmarkWorker {
+
+	public static final String TYPE = "MEMORY MAPPED FILE BASED OFFHEAP";
+	
+	private FileChannel fc;
+	
+	public MemoryMappedFileBasedOffHeapBenchmarkWorker(int elementCount) {
+		super(elementCount, TYPE);
+	}
+	
+	@Override
+	protected ByteBuffer createByteBuffer(int size) {
+		try {
+			fc = 
+				new RandomAccessFile(
+					new File("OffHeapBenchmarkTrade.bin"), "rw").getChannel();
+			return fc.map(FileChannel.MapMode.READ_WRITE, 0, size);
+		}
+		catch (Throwable t) {
+			logger.error("Unable to initialize memory mapped file !", t);
+			if (fc != null) {
+				try {
+					fc.close();
+				} 
+				catch (IOException e) {
+					logger.error(e);
+				}
+			}
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 32:39, 51:58.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.